engine/src/main/java/org/terasology/rendering/dag/nodes/CopyToVRFrameBuffersNode.java

```diff
@@ -0,0 +1,121 @@
+package org.terasology.rendering.dag.nodes;
+
+import org.terasology.rendering.dag.stateChanges.BindFBO;
+import org.terasology.rendering.openvrprovider.OpenVRProvider;
+import org.terasology.rendering.openvrprovider.OpenVRStereoRenderer;
+import org.lwjgl.opengl.GL11;
+import org.terasology.config.Config;
+import org.terasology.config.RenderingConfig;
+import org.terasology.config.RenderingDebugConfig;
+import org.terasology.monitoring.PerformanceMonitor;
+import org.terasology.registry.In;
+import org.terasology.rendering.assets.material.Material;
+import org.terasology.rendering.dag.AbstractNode;
+import static org.terasology.rendering.opengl.DefaultDynamicFBOs.FINAL;
+import org.terasology.rendering.opengl.ScreenGrabber;
+import org.terasology.rendering.opengl.fbms.DisplayResolutionDependentFBOs;
+import org.terasology.rendering.world.WorldRenderer;
+import org.terasology.rendering.world.WorldRenderer.RenderingStage;
+import static org.lwjgl.opengl.GL11.GL_COLOR_BUFFER_BIT;
+import static org.lwjgl.opengl.GL11.GL_DEPTH_BUFFER_BIT;
+import static org.lwjgl.opengl.GL11.glClear;
+import static org.terasology.rendering.opengl.OpenGLUtils.renderFullscreenQuad;
+
+public class CopyToVRFrameBuffersNode extends AbstractNode {
+    private OpenVRProvider vrProvider;
+    private OpenVRStereoRenderer vrRenderer;
+
+    @In
+    private WorldRenderer worldRenderer;
+
+    @In
+    private Config config;
+
+    @In
+    private ScreenGrabber screenGrabber;
+
+    @In
+    private DisplayResolutionDependentFBOs displayResolutionDependentFBOs;
+
+    private RenderingDebugConfig renderingDebugConfig;
+    private RenderingConfig renderingConfig;
+    private Material finalPost;
+    private Material debug;
+
+    public void setOpenVRProvider(OpenVRProvider providerToSet) {
+        this.vrProvider = providerToSet;
+    }
+
+
+    @Override
+    public void initialise() {
+        renderingConfig = config.getRendering();
+        renderingDebugConfig = renderingConfig.getDebug();
+
+        finalPost = worldRenderer.getMaterial("engine:prog.post");
+        debug = worldRenderer.getMaterial("engine:prog.debug");
+        addDesiredStateChange(new BindFBO(FINAL));
+    }
+
+    @Override
+    public void process() {
+        PerformanceMonitor.startActivity("rendering/copyToVRFrameBuffers");
+        if (!renderingDebugConfig.isEnabled()) {
+            finalPost.enable();
+        } else {
+            debug.enable();
+        }
+
+        if (!renderingConfig.isVrSupport()) {
+            logger.warn("CopyToVRFrameBufferNode processed in non-VR mode.");
+        } else {
+            renderFinalStereoImage(worldRenderer.getCurrentRenderStage());
+        }
+
+        PerformanceMonitor.endActivity();
+    }
+
+    private void renderFinalStereoImage(RenderingStage renderingStage) {
+        if (this.vrRenderer == null) {
+            this.vrProvider.init();
+            this.vrRenderer = new OpenVRStereoRenderer(this.vrProvider, 1280, 720);
+        }
```

Comment:
For some reason, when I tried to include this in the initialization the frame buffers weren't being properly created.  In hindsight, I think that could have been due to an earlier issue I had where `updatePoses()` (an OpenVR API method) was being called more often than `submitFrame()` in a separate thread (my initial design was to have a separate poller for controllers/poses, but OpenVR didn't seem to like that, and the documentation suggests it was a a flawed approach; OpenVR wants `updatePoses()` to be called on the same thread as `'submitFrame()`). At any rate, I will try to work it back into the initialization.

The 1280 x 720 frame buffer size just happens to be the size that the SDK example uses.
https://github.com/ValveSoftware/openvr/blob/master/samples/hellovr_opengl/hellovr_opengl_main.cpp

I will change make the frame buffer size adjustable and take out the hard coding.